src/Symfony/Component/Console/Application.php

```diff
@@ -1016,6 +1023,22 @@ private function findAlternatives($name, $collection)
     public function setDefaultCommand($commandName)
     {
         $this->defaultCommand = $commandName;
+
+        return $this;
+    }
+
+    public function setSingleCommand($commandName)
+    {
+        if (null !== $this->singleCommand) {
+            throw new \LogicException('A Single command is already defined.');
```

Comment:
I thinks it's useful restriction, yet it makes this feature less flexible.
And on framework level, I think it should be that way.

Strict code should be in app/domain level